### PR TITLE
Adapt webpack-rtl-plugin to our repo structure

### PR DESCRIPTION
**Pull request by scinos**
### Background

See #47451 

#### Changes proposed in this Pull Request

* Clean up `package.json` and rename plugin
* Update dependencies
* Migrate tests from `jest` to `mocha`
* Use `@automattic/webpack-rtl-plugin` in `calypso-build`

#### Testing instructions

* Checkout trunk and build the project with `yarn build-client-both`. This will produce `public/evergreen` and `public/fallback`. Save those folders somewhere, this is your baseline
* Checkout this branch, install dependencies (`yarn distclean && yarn`), and run `yarn build-client-both` again. Compare the new `public/evergreen` and `public/fallback` with the baselines from the previous step. There should be no changes.

### CODE REVIEW

**Review by jsnajdr — COMMENTED** [review]
The package's directory also has its own `yarn.lock`, `.gitignore` and `.npmignore`. Can we also delete them? The information in `.npmignore` could be replaced by a `files` field in `package.json`.

I'm not sure about deleting the `LICENSE.md` file. After we publish the package to NPM, the root `CREDITS.md` file is no longer anywhere near.

One idea for future improvement (not in this PR): make `cssnano` a peer dependency, so that upgrade of `cssnano` is decoupled from upgrade of this plugin.

File: packages/webpack-rtl-plugin/package.json [hunk 1]
```diff
@@ -1,38 +1,36 @@
 {
-	"name": "webpack-rtl-plugin",
+	"name": "@automattic/webpack-rtl-plugin",
 	"version": "2.0.0",
-	"description": "Webpack plugin to produce a rtl css bundle",
+	"description": "Webpack plugin to produce a rtl css bundle. Forked from https://github.com/romainberger/webpack-rtl-plugin",
```
Comment (jsnajdr) [inline packages/webpack-rtl-plugin/package.json]: Personal opinion: I'd rather keep the description short and one sentence. We can credit the original author somewhere else in a better way.

File: packages/webpack-rtl-plugin/test/index.js [hunk 2]
```diff
@@ -183,15 +182,15 @@ describe( 'Webpack RTL Plugin', () => {
...
+			expect( fs.existsSync( cssBundlePath ) ).toBe( true );
+			expect( fs.existsSync( rtlCssBundlePath ) ).toBe( true );
 		} );
 
 		it( 'should create a second bundle with a different hash', () => {
 			const cssChunk = cssBundleName.split( '.' );
 			const rtlCssChunk = rtlCssBundleName.split( '.' );
 
-			expect( cssChunk[ 1 ] ).to.not.equal( rtlCssChunk[ 1 ] );
+			expect( cssChunk[ 1 ] ).not.toEqual( rtlCssChunk[ 1 ] );
```
Comment (jsnajdr) [inline packages/webpack-rtl-plugin/test/index.js]: `.toBe` is enough to compare strings, `.toEqual` is deep comparison.

File: packages/webpack-rtl-plugin/test/index.js [hunk 3]
```diff
@@ -247,25 +246,23 @@ describe( 'Webpack RTL Plugin', () => {
...
+			expect( fs.existsSync( cssBundlePath ) ).toBe( true );
+			expect( fs.existsSync( rtlCssBundlePath ) ).toBe( true );
 		} );
 
 		it( 'should create a second bundle with a different hash', () => {
 			const cssChunk = cssBundleName.split( '.' )[ 1 ];
 			const rtlCssChunk = rtlCssBundleName.split( '-' )[ 2 ];
 
-			expect( cssChunk ).to.not.equal( rtlCssChunk );
+			expect( cssChunk ).not.toEqual( rtlCssChunk );
```
Comment (jsnajdr) [inline packages/webpack-rtl-plugin/test/index.js]: `.toBe`

File: packages/webpack-rtl-plugin/test/index.js [hunk 4]
```diff
@@ -500,14 +495,14 @@ describe( 'Webpack RTL Plugin', () => {
 			const expected = fs
 				.readFileSync( path.join( __dirname, 'rtl-diff-result.css' ), 'utf-8' )
 				.replace( /\r/g, '' );
-			expect( contentRrlCss ).to.equal( expected );
+			expect( contentRrlCss ).toEqual( expected );
```
Comment (jsnajdr) [inline packages/webpack-rtl-plugin/test/index.js]: `.toBe`

File: packages/webpack-rtl-plugin/package.json (hunk 1 above)
Comment (scinos) [reply packages/webpack-rtl-plugin/package.json]: Fixed

File: packages/webpack-rtl-plugin/test/index.js (hunk 2 above)
Comment (scinos) [reply packages/webpack-rtl-plugin/test/index.js]: Fixed

File: packages/webpack-rtl-plugin/test/index.js (hunk 4 above)
Comment (scinos) [reply packages/webpack-rtl-plugin/test/index.js]: Fixed

File: packages/webpack-rtl-plugin/test/index.js (hunk 3 above)
Comment (scinos) [reply packages/webpack-rtl-plugin/test/index.js]: Fixed

**Review by scinos — COMMENTED** [review]
> The package's directory also has its own yarn.lock, .gitignore and .npmignore. Can we also delete them? The information in .npmignore could be replaced by a files field in package.json.

`yarn.lock` and `.npmignore` deleted. I left `.gitignore` because we have a few packages with a `.gitignore` in them, and IMO is more clear and less likely to get out of sync than a monolithic `.gitignore` in the root.

> I'm not sure about deleting the LICENSE.md file. After we publish the package to NPM, the root CREDITS.md file is no longer anywhere near.

Good point. I restored the original LICENSE.md. After searching a bit looks like we can relicense it as GPL to match the rest of the repo (sources: https://en.wikipedia.org/wiki/MIT_License, https://www.gnu.org/licenses/license-list.html#Expat, https://www.gnu.org/licenses/gpl-faq.html#WhatDoesCompatMean)

**Review by jsnajdr — APPROVED** [review]
Looks good :+1:

**Pull request merged**